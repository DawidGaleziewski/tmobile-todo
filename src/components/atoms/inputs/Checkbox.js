import React from "react";

// Libraries
import { css, jsx } from "@emotion/core";

/** @jsx jsx */
const labelStyle = (value) => css`
  background-color: ${value ? "green" : "#db2828"};
  padding: 0.6rem 1.2rem;
  border-radius: 16px;
  color: #fff;
  cursor: pointer;
  font-size: 1.2rem;
  display: inline-block;
  /* opacity: 0.6; */
  -webkit-transition: 0.3s ease-in all;
  transition: 0.3s ease-in all;
  white-space: nowrap;
`;
const inputStyle = css`
  display: none;
`;

const Checkbox = ({ id, name, value, onChange }) => {
  return (
    <div className="input-slider-wrapper">
      <input
        onChange={onChange}
        css={inputStyle}
        id={id}
        type="checkbox"
        name={name}
        value={value}
        placeholder=" "
      />
      <label css={labelStyle(value)} htmlFor={id}>
        {value ? "done" : "in progress"}
      </label>
    </div>
  );
};

export default Checkbox;
