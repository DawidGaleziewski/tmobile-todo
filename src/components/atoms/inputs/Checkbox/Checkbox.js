import React from "react";

// Libraries
import { css, jsx } from "@emotion/core";

// Styles
import { labelStyle, inputStyle } from "./style/CheckboxStyle";

/** @jsx jsx */

const Checkbox = ({ id, name, value, onChange }) => {
  return (
    <div className="input-slider-wrapper" data-testid="checkbox">
      <input
        onChange={onChange}
        css={inputStyle}
        id={id}
        type="checkbox"
        name={name}
        value={value}
        placeholder=" "
      />
      <label css={labelStyle(value)} htmlFor={id}>
        {value ? "done" : "in progress"}
      </label>
    </div>
  );
};

export default Checkbox;
