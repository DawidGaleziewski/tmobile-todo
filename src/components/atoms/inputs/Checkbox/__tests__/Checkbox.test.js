// Libraries
import React, { useContext } from "react";
import ReactDOM from "react-dom";
import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

// Components
import Checkbox from "../Checkbox";

afterEach(cleanup);

it("renders checkbox without crashing", () => {
  const divContainer = document.createElement("div");
  ReactDOM.render(<Checkbox />, divContainer);
  ReactDOM.unmountComponentAtNode(divContainer);
});

it("renders ToggleStatusBadge with correct text when checked", () => {
  const { getByTestId } = render(
    <Checkbox
      id="form-input-id"
      name="task-status"
      value={true}
      onChange={() => null}
    />
  );
  expect(getByTestId("checkbox")).toHaveTextContent("done");
});

it("renders ToggleStatusBadge with correct text when not checked", () => {
  const { getByTestId } = render(
    <Checkbox
      id="form-input-id"
      name="task-status"
      value={false}
      onChange={() => null}
    />
  );
  expect(getByTestId("checkbox")).toHaveTextContent("in progress");
});

it("matches snapshot 1", () => {
  const taskMockData = {
    id: 20,
    done: false,
    title: "test 2",
    description: "test2",
  };
  const DOMTree = renderer
    .create(
      <Checkbox
        id="form-input-id"
        name="task-status"
        value={false}
        onChange={() => null}
      />
    )
    .toJSON();

  expect(DOMTree).toMatchSnapshot();
});
