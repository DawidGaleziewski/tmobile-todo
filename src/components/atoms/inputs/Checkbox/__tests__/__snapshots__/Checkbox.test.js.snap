// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 1`] = `
<div
  className="input-slider-wrapper"
  data-testid="checkbox"
>
  <input
    className="css-cg9o93-Checkbox"
    id="form-input-id"
    name="task-status"
    onChange={[Function]}
    placeholder=" "
    type="checkbox"
    value={false}
  />
  <label
    className="css-174x1gu-Checkbox"
    htmlFor="form-input-id"
  >
    in progress
  </label>
</div>
`;
